2.cellprofiling/scripts/colocalization.py

```diff
@@ -33,10 +40,100 @@
 warnings.filterwarnings("ignore", category=RuntimeWarning)
 
 
-# In[2]:
+# In[ ]:
+
+
+def process_combination(args, image_set_loader):
+    """
+    Process a single combination of compartment and channel pair for colocalization analysis.
+
+    Parameters
+    ----------
+    args : tuple
+        A tuple containing the compartment, channel1, and channel2.
+        Yes, order matters.
+        args = (compartment, channel1, channel2)
+        Where:
+        compartment : str
+            The compartment to process.
+        channel1 : str
+            The first channel to process.
+        channel2 : str
+            The second channel to process.
+
+    image_set_loader : ImageSetLoader
+        An instance of the ImageSetLoader class that loads the images and metadata.
+
+    Returns
+    -------
+    str
+        A message indicating the completion of processing for the given combination.
+    """
+    compartment, channel1, channel2 = args
+    coloc_loader = TwoObjectLoader(
+        image_set_loader=image_set_loader,
+        compartment=compartment,
+        channel1=channel1,
+        channel2=channel2,
+    )
+    list_of_dfs = []
+    for object_id in coloc_loader.object_ids:
+        cropped_image1, cropped_image2 = prepare_two_images_for_colocalization(
+            label_object1=coloc_loader.label_image,
+            label_object2=coloc_loader.label_image,
+            image_object1=coloc_loader.image1,
+            image_object2=coloc_loader.image2,
+            object_id1=object_id,
+            object_id2=object_id,
+        )
+        colocalization_features = measure_3D_colocalization(
+            cropped_image_1=cropped_image1,
+            cropped_image_2=cropped_image2,
+            thr=15,
+            fast_costes="Accurate",
+        )
+        coloc_df = pd.DataFrame(colocalization_features, index=[0])
+        coloc_df.columns = [
+            f"Colocalization_{compartment}_{channel1}.{channel2}_{col}"
+            for col in coloc_df.columns
+        ]
+        coloc_df.insert(0, "object_id", object_id)
+        coloc_df.insert(1, "image_set", image_set_loader.image_set_name)
```

Comment:
Consider performing this work before casting into a DataFrame. It looks like `measure_3D_colocalization` provides you with a `Dict[str, float]` so this could be a matter of iterating the keys of the dict.